### ISSUE TITLE
Settings cannot be saved in linux mint for flatpack

### ISSUE THREAD

**Issue by taltamir** [crow-translate/crow-translate#599]
**Bug description**
Settings cannot be saved in linux mint for flatpack.

**Steps to reproduce**
Install and run crow translate.
Either
A. make changes in main window then restart the program. all changes lost
B. open settings and make any changes there. hit ok. get error and changes you made to settings are lost.

**Expected behavior**
Settings getting saved.

**Screenshots**
![crow translate](https://github.com/crow-translate/crow-translate/assets/7625366/c61d0378-1fe2-4b6f-abb9-dfe1aaa04d35)

**Environment**
- OS: Linux Mint
- Desktop environment: 21.2 Cinnamon
- Crow Translate version: 2.10.8
- Installed as Flatpack from Flathub, via built in GUI software manager

**Comment by Shatur** [comment]
It looks like the app is trying to register itself for autostart but is failing because flatpak doesn't allow it.
@gasinvein looks like it fails to detect that the app is inside the flatpak?

Also I would recommend to update to 2.10.10 because 2.10.8 have broken selection translation.

**Comment by gasinvein** [comment]
> The app is trying to register itself for autostart but is failing because flatpak doesn't allow it.
> @gasinvein looks like it fails to detect that the app is inside the flatpak?

Nah, it tries to communicate with the background portal (which means it has properly detected flatpak sandbox) but fails.

@taltamir is `xdg-desktop-portal` installed and running? What version is it?

> Also I would recommend to update to 2.10.10 because 2.10.8 have broken selection translation.

Ugh, I guess we should just allow successful update builds to be published automatically.

**Comment by Shatur** [comment]
> Ugh, I guess we should just allow successful update builds to be published automatically.

Sounds awesome. Should we automate it using GitHub Actions?

**Comment by gasinvein** [comment]
> > Ugh, I guess we should just allow successful update builds to be published automatically.
> 
> Sounds awesome. Should we automate it using GitHub Actions?

No, Flathub's update bot already [submits](https://github.com/flathub/io.crow_translate.CrowTranslate/pulls?q=is%3Apr+author%3Aflathubbot+is%3Amerged+) PRs following upstream project updates. It's just a matter of flipping a setting to tell the bot that said PRs should be also merged on successful build.

**Comment by Shatur** [comment]
Oh, got it. Sounds like a reasonable choice.

**Comment by taltamir** [comment]
> @taltamir is `xdg-desktop-portal` installed and running? What version is it?

thank you. it is installed. version: 
1.14.4-1ubuntu2~22.04.1

> Ugh, I guess we should just allow successful update builds to be published automatically.

That kind of automation does make a lot of sense.
It updated crow to v2.10.10

It still gives me that error when changing anything in the settings menu. However, it now correctly auto saves main window changes between sessions.
By main window settings I mean clicking the box "auto translate" and adding the languages hebrew and english to both the source and the destination quicklist.

Before today's update to crow those settings were lost each time I restarted. Now they are preserved.
... although. thinking about it, it might have been because I actually restarted the machine before. While this time I ordered crow to quit first. giving it an opportunity to save those changes in a way that restarting mint would not.

let me go test this real quick by adding another language and then calling a machine restart while crow remains running

**Comment by taltamir** [comment]
I confirmed that the problem remains. I just added afrikans to destination languages on main interface and then restart the OS and the change was lost on system reboot.

It now appears to me that I have reported 2 separate bugs.

Bug 1: can't change settings in the settings menu (not just the launch on boot setting. any setting I tried in the settings menu gives that error)

Bug 2: language changes on main interface are saved correctly. but only if when and if you use the [Quit] button in the system tray. When restarting the OS those changes are lost as the [Quit] button was not pressed and thus did not trigger save.
I honestly think those should be saved as soon as they are made rather than waiting for the [Quit] button to be pressed.

**Comment by gasinvein** [comment]
> > @taltamir is `xdg-desktop-portal` installed and running? What version is it?
> 
> thank you. it is installed. version: 1.14.4-1ubuntu2~22.04.1

It appears that 1.14 is too old and doesn't provide the required interface, see flatpak/xdg-desktop-portal#1014

**Comment by taltamir** [comment]
> It appears that 1.14 is too old and doesn't provide the required interface, see [flatpak/xdg-desktop-portal#1014](https://github.com/flatpak/xdg-desktop-portal/issues/1014)

thank you. I am looking into it further on mint side of things to get mint updated to v1.16 (latest official release).

I will also investigate some other linux distros and see when they switched (or if they switched) from 1.14 to v1.16.

out of curiosity. what method did crow translate use before 1.16? as that version is only from 4 months ago.
if a bunch of linux distros are affected it might be worthwhile to probe the version of `xdg-desktop-portal` and fallback to the old method in case 1.16 is unavailable.

hmm... or maybe, can flathub be set to note such compatibility problems and automatically use an older version of crow translate on affected OS? Or does flathub only keep a single latest version?
I looked into it casually a bit and I am not seeing a method to do this via flatpack. but maybe there is and I am just ignorant.

**Comment by taltamir** [comment]
> In flatpak Crow Translate didn't have a working autostart method in before the portal one.

What about the rest of the settings menu? It is not just autostart, any setting you try to change fails
